spec/classes/pulpcore_spec.rb

```diff
@@ -481,6 +481,66 @@
             ])
         end
       end
+
+      context 'with 24 processors from facts and default worker_count' do
+        let(:facts) { override_facts(os_facts, processors: {count: 24}) }
+
+        it 'enables only 8 pulpcore workers' do
+          (1..8).to_a.each do |i|
```

Comment:
Do you need to use `.to_a` here? I think this also works
```suggestion
          (1..8).each do |i|
```
(same below)

Reply:
Thanks! I tested this and it works just as well.

It's been a while since I was first exposed to the concept of "Duck Typing" in Ruby, which I guess usually means using `object.method if object.responds_to?(method)` instead of `object.method if object.is_a?(specific_class_that_implements_method)`.

But under that same concept I would also include an understanding that if multiple classes implement the "same" method, and an object can be converted back and forth between the two classes without any loss of data (or if the conversion can be performed one-way without any loss of data which is relevant to that particular method, i.e. if the conversion is an "injective morphism" to borrow a term from math) then broadly speaking you should expect the same result whether calling the method on the initial object or converting the object and then calling the method. And in that case, there's no point in performing the conversion.... just call the method on the first object.

This is all an unnecessarily long winded way of saying: thanks for the suggestion, I found this to be not only useful for improving this code, but also generally kind of neat and led me to an understanding which should make me a better programmer in the longer term.